app/scripts/lib/rpc-method-middleware/handlers/handlers-helper.ts

```diff
@@ -0,0 +1,22 @@
+export type HandlerWrapperType = {
+  methodNames: [string];
+  hookNames: Record<string, boolean>;
+};
+
+/**
+ * @typedef {object} ProviderStateHandlerResult
```

Comment:
```suggestion
```

Reply:
Based the branch to https://github.com/MetaMask/metamask-extension/pull/23589 so that it will be easier to keep track of `handlers-helper.ts` along all the files that use it.